File: booking-house-backend/app/Models/UserModel.js
const { DataTypes } = require('sequelize');
const sequelize = require('../../config/database');

const User = sequelize.define('NguoiDung', {
    nguoi_dung_id: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true
    },
    ho_ten: {
        type: DataTypes.STRING,
        allowNull: false
    },
    email: {
        type: DataTypes.STRING,
        allowNull: false,
        unique: true
    },
    mat_khau: {
        type: DataTypes.STRING,
        allowNull: false
    },
    so_dien_thoai: {
        type: DataTypes.STRING
    },
    ngay_tao: {
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW
    },
    role_id: {
        type: DataTypes.INTEGER
    }
}, {
    tableName: 'nguoi_dung',
    timestamps: false
});

module.exports = User;

File: bookinghouse/lib/models/user_model.dart
class User {
  final int id;
  final String fullName;
  final String email;
  final String? phone;
  final int? roleId;

  User({
    required this.id,
    required this.fullName,
    required this.email,
    this.phone,
    this.roleId,
  });

  factory User.fromJson(Map<String, dynamic> json) {
    return User(
      id: json['id'] ?? json['nguoi_dung_id'] ?? 0,
      fullName: json['ho_ten'] ?? '',
      email: json['email'] ?? '',
      phone: json['so_dien_thoai'],
      roleId: json['role_id'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'nguoi_dung_id': id,
      'ho_ten': fullName,
      'email': email,
      'so_dien_thoai': phone,
      'role_id': roleId,
    };
  }
}
